src/build.d

```diff
@@ -467,9 +459,9 @@ alias html = makeDep!((htmlBuilder, htmlDep) {
             .sources(sourceArray)
             .target(env["DOC_OUTPUT_DIR"].buildPath(d2html(source)[srcDir.length + 1..$]
                 .replace(dirSeparator, "_")))
-            .deps([versionFile, sysconfDirFile])
+            .deps([dmdDefault, versionFile, sysconfDirFile])
             .command([
-                env["DMD"],
+                dmdExe(null, null).target,
```

Comment:
I modified it to use `dmdDefault.deps[0].target`.  This does rely on `dmdExe` being first dependency, but, if this ever changes than it should break the build as it would mean you'd be trying to generate the html docs with a file other than the compiler executable.

Reply:
I think it would be favorable to use `dmdDefault.target` here and add `target(dep.deps[0])` to `dmdDefault`.

(Kinda pedantic but this avoids distributing knowledge about `dmdDefault` to multiple places)

Reply:
`dmdDefault` doesn't have any targets. It has 2 dependencies.

Reply:
Usually not but `dmdDefault` is just a dependency group

Maybe consider another perspective: Why does `html` depend on `dmdDefault` as it only uses a single dependency? (=> because DMD requires a configuration file).

I think this violates the Principle of Least Surprise and requires that this pattern is applied to all targets using the generated compiler

Reply:
In all cases, the target files represent the files being generated by the command of the dependency. The target file is also used to determine whether the command needs to be run. Having the same target in 2 different dependencies makes it alot more complicated.  I'd rather write `dmdDefault.deps[0].target` than complicate the way dependencies work and get to write `dmdDefault.target`.

Reply:
Okay.